connectors/src/admin/cli.ts

```diff
@@ -386,6 +386,121 @@ const notion = async (command: string, args: parseArgs.ParsedArgs) => {
 
       return;
     }
+
+    case "batch-resync": {
+      if (!args.size) {
+        throw new Error("Missing --size argument");
+      }
+
+      if (!["small", "medium", "large"].includes(args.size)) {
+        throw new Error("--size must be small, medium or large");
+      }
+
+      // 0-indexed
+      let batchNumber: number | null = null;
+      let batchSize: number | null = null;
+
+      if (args.batchNumber) {
+        batchNumber = parseInt(args.batchNumber as string, 10);
+      }
+
+      if (args.batchSize) {
+        batchSize = parseInt(args.batchSize as string, 10);
+      }
+
+      if (args.size !== "small" && (!batchNumber || !batchSize)) {
+        throw new Error(
+          `--batchNumber and --batchSize are required for size ${args.size}`
+        );
+      }
+
+      if ((batchNumber === null) !== (batchSize === null)) {
+        throw new Error(
+          `--batchNumber and --batchSize must be both specified or both null`
+        );
+      }
+
+      const queryRes = await sequelize_conn.query(`
+        SELECT COUNT(*) c, "connectorId" 
+        FROM notion_pages
+        GROUP BY "connectorId"
+        ORDER BY "connectorId" ASC`);
+
+      const connectorIdsWithCount = queryRes[0] as {
+        c: number;
+        connectorId: string;
+      }[];
+
+      const filter = (c: number) => {
+        if (args.size === "small") {
+          return c <= 10_000;
+        }
+
+        if (args.size === "medium") {
+          return c > 10_000 && c <= 60_000;
+        }
+
+        if (args.size === "large") {
+          return c > 60_000;
+        }
+      };
+
+      const connectorIds = connectorIdsWithCount
+        .filter((c) => filter(c.c))
+        .map((c) => c.connectorId);
+
+      let connectorsToResync = [];
+
+      if (batchNumber && batchSize) {
+        const start = batchNumber * batchSize;
```

Comment:
nit: a bit brittle since connectors can be added / deleted while we run those (not a big issue now but later on maybe?)
probably not a biggie
=> connector deleted means we'll redo a full sync for one connector
=> connector added will just mean we have to be careful in the rare case it creates another batch, to remember to run it